feature-libs/cart/base/components/add-to-cart/add-to-cart.component.ts

```diff
@@ -121,24 +128,29 @@ export class AddToCartComponent implements OnInit, OnDestroy {
     if (this.product) {
       this.productCode = this.product.code ?? '';
       this.setStockInfo(this.product);
-      this.cd.markForCheck();
     } else if (this.productCode) {
       // force hasStock and quantity for the time being, as we do not have more info:
       this.quantity = 1;
       this.hasStock = true;
       this.cd.markForCheck();
     } else {
+      let productObservable$: Observable<Product | null>;
```

Comment:
[NITPICK]
`productObservable$` vs `product$`
(`$` already indicates  - by convention - that it's an observable)

Reply:
Renamed